CMSgov/mint-app
# [EASI-2656] - Operational solution details content

# EASI-2656

## Changes and Description

- Updated modal routing to utilize query params instead of route param
- Added generic components for `About`, `Timeline`, and `Points of contact`
- Added custom content for certain complex timelines
- `GatheringInfoAlert` for solutions with content yet to come
- Deprecated `usePreviousModalRoute` in favor of query params

![Screenshot 2023-03-03 at 2 13 35 PM](https://user-images.githubusercontent.com/95709965/222806905-3343a69c-24d6-4f10-9789-057f4b5dfd1f.png)


## How to test this change

- Add yourself as target to the feature flag `operationalSolutionHelp`
- Verify that modal renders correct content
- Verify that scroll positions, previous route content, and params are preserve when opening and closing modal

## Notes

- Added small UI fix for milestones datepicker

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
